posthog/test/test_migration_0220.py

```diff
@@ -0,0 +1,84 @@
+from posthog.test.base import TestMigrations
+
+
+class TagsTestCase(TestMigrations):
+
+    migrate_from = "0219_migrate_tags_v2"  # type: ignore
+    migrate_to = "0220_backfill_primary_dashboards"  # type: ignore
+    assert_snapshots = True
+
+    def setUpBeforeMigration(self, apps):
+        Organization = apps.get_model("posthog", "Organization")
+        Dashboard = apps.get_model("posthog", "Dashboard")
+        Team = apps.get_model("posthog", "Team")
+
+        org = Organization.objects.create(name="o1")
+
+        # CASE 1:
+        # Team with an existing primary dashboard, and non primary dashboards
+        # Expect: t1.primary_dashboard = d2 (even though d1 is created first)
+        team_1 = Team.objects.create(name="t1", organization=org)
+        Dashboard.objects.create(name="d1", team=team_1)
+        dashboard_2 = Dashboard.objects.create(name="d2", team=team_1)
+        team_1.primary_dashboard = dashboard_2
+        team_1.save()
+
+        # CASE 2:
+        # Team with no dashboards
+        # Expect: t2.primary_dashboard = None
+        Team.objects.create(name="t2", organization=org)
+
+        # CASE 3:
+        # Team with multiple pinned and non pinned dashboards, and no primary set
+        # Expect: t3.primary_dashboard = d4 (even though d3 is created first)
+        team_3 = Team.objects.create(name="t3", organization=org)
+        Dashboard.objects.create(name="d3", team=team_3, pinned=False)
+        Dashboard.objects.create(name="d4", team=team_3, pinned=True)
+        Dashboard.objects.create(name="d5", team=team_3, pinned=True)
+
+        # CASE 4:
+        # Team with multiple non-pinned dashboards, and no primary set
+        # Expect: t4.primary_dashboard = d6
+        team_4 = Team.objects.create(name="t4", organization=org)
+        Dashboard.objects.create(name="d6", team=team_4)
+        Dashboard.objects.create(name="d7", team=team_4)
+
+        # BATCH CASE
+        teams = Team.objects.bulk_create(
+            [Team(name=f"batch_team-{team_number+10}", organization=org) for team_number in range(501)]
+        )
+        Dashboard.objects.bulk_create(
+            [
+                Dashboard(name=f"batch_dashboard-{dashboard_number+10}", team=team,)
+                for dashboard_number, team in enumerate(teams)
+            ]
+        )
+
+    def test_backfill_primary_dashboard(self):
+        Dashboard = self.apps.get_model("posthog", "Dashboard")  # type: ignore
+        Team = self.apps.get_model("posthog", "Team")  # type: ignore
+
+        # CASE 1:
```

Comment:
100% agree, I think it would take a bit of refactoring on the `TestMigrations` class to make that happen (which then impacts the other migrations using that class), and I was trying to keep this PR scoped down - hence, the comments trying to link the setup + tests